llvm/lib/CodeGen/MachineLICM.cpp

```diff
@@ -1249,6 +1249,15 @@ bool MachineLICMBase::IsProfitableToHoist(MachineInstr &MI,
     return false;
   }
 
+  // If we have a COPY with other uses in the loop, hoist to allow the users to
+  // also be hoisted.
+  if (MI.isCopy() && IsLoopInvariantInst(MI, CurLoop) &&
```

Comment:
Defer the IsLoopInvariantInst check until after the cheaper register checks

Reply:
Alright.